### DISCUSSION TITLE
2024-06-28 - Clean Architecture Template - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by bradystroud** [SSWConsulting/SSW.Rules.Content#8794] [Chewing the Fat]
Lets discuss the SSW Clean Architecture template.

https://github.com/SSWConsulting/SSW.CleanArchitecture

Have you used the template before?
Did it save you lots of effort?

**Comment by PothieuG** [comment]
I really appreciate the efforts made to provide a template that is so easy, quick, and comprehensive to set up. 
I used the old template for a project, which was already great at the time. But this one seems even more amazing!

**Comment by bahjatalaadelssw** [comment]
I've known about the template for ages because I edit the SAW team meeting updates. But I'm not a dev and have not used it.

**Comment by ozdemir-mehmet** [comment]
Good topic to finish off the week.

**Reply by jakebayliss** [reply]
Cheers Mehmet

**Comment by bettybondoc** [comment]
Not an SSW Dev, haven't used it before but looks interesting!

**Comment by MarcusIrmscherSSW** [comment]
Great video Dan, you're a natural ⭐

**Comment by stephanfakossw** [comment]
I can see how this can be interesting to a dev...
For me, it's an N/a - not a dev

**Comment by leahy268** [comment]
Great video Dan.   Not qualified enough to comment on how good the Clean Architecture Template is though.

**Comment by calumjs** [comment]
Looks like an awesome template, nice work!
Based on the previous week's architectural decision tree however, CA will no longer be recommended for the majority of projects.
So I think we should focus our efforts on the architectures that are likely to be recommended e.g. VSA

Also good to see we have returned to the ascending chart emoji 📈

**Comment by jakebayliss** [comment]
Have set up a new project with the older versions of our CA template and it was easy enough to set up but annoying to remove all the bits I didnt need/want. The commands to generate a new query or command are really cool and I'll definitely be using that

**Comment by ricksu978** [comment]
> Have you used the template before?

Never use a template before.
When starting a new CA project, I like to handcraft the project and try different arrangement.

**Comment by joshbermanssw** [comment]
great run through of how to use the template!

**Comment by ncn-ssw** [comment]
> Have you used the template before?

No - but I could see myself using it as a reference in future

> Did it save you lots of effort?

N/A

**Comment by jimmy1992abc** [comment]
> Have you used the template before?
> Did it save you lots of effort?

No - Not a developer.

The video is great, simple and clear. I understand every word, unfortunately less knowledge after putting them together😆

**Comment by MattParkerDev** [comment]
Template looks awesome, thanks to the team for continuously improving it :)

**Comment by Calinator444** [comment]
> Have you used the template before?


I've meddled around with the Clean Architecture template a bit and made a video, but I made this when I had a naive understanding of what Clean Architecture was. I had some false assumptions about the design pattern, such as the assumption that CQRS and DDD were features of CA rather than just how CA was implemented in the template.

See my done video https://youtu.be/J-mb4b-tuqA 


> Did it save you lots of effort?

No, because I was just setting up the project. In the context of a large-scale project having an architecture pattern in place would help you down the line when you need to start expanding and adding new features to your project.

**Comment by YalalCakirSSW** [comment]
I attended Clean Architecture Superpowers couple weeks ago which was super informative. Many thanks to @danielmackay & @Hona 

> Have you used the template before?

We are using a version of CA in TimePro. 

> Did it save you lots of effort?

Can't compare, TimePro was already using CA when I joined so I'm not familiar with the prev effort.

**Reply by Calinator444** [reply]
I figured you'd enjoy the template. You were the only other person in FBC who implemented global exception handling in their project.

**Comment by awaltos** [comment]
I have never used the template but it seems like a great starting off point especially for someone new to CA and dotnet.

**Comment by cvionnet** [comment]
Thanks to the team for maintaining the template! 🚀

> Have you used the template before?

No, but this is definitively something I will use when starting a project from scratch

> Did it save you lots of effort?

Not yet

**Comment by christoment** [comment]
Having a template (CleanArchitecture repo) and a reference project (Northwind) both serves really well as a guideline when working on related projects 😄 

> Have you used the template before?

Use the new template in a project? Not yet

> Did it save you lots of effort?

I'm hoping to try and get my findings 😄
But definitely a lot of things in this template I've been reusing in many projects (e.g. how to properly set up TestContainer + Respawn - copy paste is way easier than having to set all the fixtures manually!)

**Comment by matt-goldman-ssw** [comment]
> Have you used the template before?
Did it save you lots of effort?

I haven't used it yet because I haven't bootstrapped a client project with CA since we've had the template. Our new template has a bunch of features that distinguish it from others and we should promote these!

**Comment by jeoffreyfischer** [comment]
I'm keen to use this template on my free time - didn't know we had one

**Comment by duanxinhuan** [comment]
good work

**Comment by chrisschultzssw** [comment]
> Have you used the template before?
> Did it save you lots of effort?

N/A - Not a dev :). Great video though!

**Comment by sethdaily** [comment]
> Have you used the template before?

No

> Did it save you lots of effort?

Yep!

**Comment by JackDevAU** [comment]
DevEx seems to be pretty good! Amazing stuff team! 

Will definitely love to get some more mentoring around DDD and Clean Code 👀 

### **Q. Are people going to start using this template to spin up new client projects?**

**Comment by Ben0189** [comment]
Have you used the template before?
No

Did it save you lots of effort?
I have been using things in this project as reference/ learning

**Comment by Marxoz** [comment]
Have you used the template before?
Yes

Did it save you lots of effort?
Defintely!

Awesome work on the Template gang!

**Comment by AntPolkanov** [comment]
> Have you used the template before?
No

> Did it save you lots of effort?
In the future, yes

**Comment by steven0x51** [comment]
Great topic! I haven't used the template for a project yet, but been using it as a reference

**Comment by Dhruv-0987** [comment]
I am loving this software architecture series, getting so much value of this and Dan is making it easier than expected 

Have you used the template before?
Yes I used or more referenced the integration testing setup to improve the integration testing setup for SugarLearning.

Did it save you lots of effort?
Yes lots

**Comment by AttackOnMorty** [comment]
Awesome video. I haven't used this template but will definitely try.

PS: The **SAW** team sounds "scary".

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/37203901/1aeb6f81-16b6-4229-a91a-bb93ee01dee9)

**Comment by zacharykeeping** [comment]
> Have you used the template before?

Not yet

> Did it save you lots of effort?

I'm sure it will in the future.

**Comment by Tanya-Leahy** [comment]
Great video, not a dev though.

**Comment by SamBWagner** [comment]
> Have you used the template before?

Nope. No plan to use it in future. Could be good, just not seeing a reason to use it for me.

> Did it save you lots of effort?

No.

**Comment by amankumarrr** [comment]
No, but I have saved the link for the future use.

**Comment by ozair-ashfaque** [comment]
> Have you used the template before?
Not yet - but definitely use it on my free time

**Comment by PennyWalker** [comment]
Great video Dan - I think this template will help a lot of people :)

**Comment by KristenHu** [comment]
Not yet, but I'll definitely try next time.

**Comment by jimmidier** [comment]
> Have you used the template before?

No, but used the old template before.

> Did it save you lots of effort?

N/A.

**Comment by jerryluossw** [comment]
> Have you used the template before?

No, but I'll probably try it in the future.

> Did it save you lots of effort?

N/A, but I think it can save lots of effort, I especially like **ssw-ca-command** and **ssw-ca-query**

**Comment by Aibono1225** [comment]
> Have you used the template before?

Not yet, but I'd like to

> Did it save you lots of effort?

N/A

**Comment by babakamyljanovssw** [comment]
> Have you used the template before?

Not yet, I will use it as reference

> Did it save you lots of effort?

N/A

**Comment by isaaclombardssw** [comment]
> Have you used the template before?

No, but it helps to know about the resource for future projects

**Comment by danielmackay** [comment]
> Have you used the template before?

No, but I have taken the best parts of it to retro fit existing projects.  Not only is it a CA template, but it also contains many other best practices around DDD and Web API development

> Did it save you lots of effort?

Yes.

**Comment by tkapa** [comment]
I've only tested the template previously. There aren't many situations in which I've had the opportunity to start using the template.

Though I can see that it would save a lot of effort in the long run.

**Comment by tino-liu** [comment]
> Have you used the template before?

Would love to try it! The video was very well made so setting it up shouldn't be too hard

> Did it save you lots of effort?
N/A

**Comment by TobyChurches** [comment]
I love the template! Although I haven't started a project from scratch in a while, Ill definitely be using this + the decision tree next time I do

**Comment by vladislav-kir** [comment]
> Have you used the template before?
Did it save you lots of effort?

I used the old template once before, and, although I was not familiar with it then, it was fairly easy to setup and understand.
